server/tests/test_get_symbol.rs

```diff
@@ -76,57 +100,178 @@ fn test_hover_on_model_field_and_method() {
+    );
+    let hover_country_id = get_hover_markdown(31, 39).unwrap_or_default();
+    assert!(
+        hover_country_id.contains("country_id: Country"),
+        "Hover on field_name in search domain should show field name and field type"
+    );
+    let hover_code = get_hover_markdown(31, 48).unwrap_or_default();
+    assert!(
+        hover_code.contains("code: str"),
+        "Hover on field_name in search domain should show field name and field type"
+    );
+
     // Hover on a variable assignment (baseInstance1)
-    let hover_var = get_hover_markdown(31, 0).unwrap_or_default();
+    let hover_var = get_hover_markdown(33, 0).unwrap_or_default();
     assert!(
         hover_var.contains("BaseTestModel"),
         "Hover on variable should show type info"
     );
 }
 
-// TODO:
-// - Hover on domains
-// - Hover on related fields parameters
-// - Hover on computed field (compute method field)
+#[test]
+fn test_definition() {
+    // Setup server and session with test addons
+    let odoo_path = env::var("COMMUNITY_PATH").unwrap();
+    let odoo_path = PathBuf::from(odoo_path).sanitize();
+    let odoo_path = odoo_path.as_str();
+
+    let mut odoo = setup::setup::setup_server(true);
+    let test_addons_path = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("tests").join("data").join("addons");
+    let module1_test_file = test_addons_path.join("module_1").join("models").join("base_test_models.py").sanitize();
+    let module2_test_file = test_addons_path.join("module_2").join("models").join("base_test_models.py").sanitize();
+
+    // Ensure the test file exists
+    assert!(PathBuf::from(&module1_test_file).exists(), "Test file does not exist: {}", module1_test_file);
+    assert!(PathBuf::from(&module2_test_file).exists(), "Test file does not exist: {}", module1_test_file);
+    let mut session = setup::setup::create_session(&mut odoo);
+
+    // Get file symbol and file info
+    let file_mgr = session.sync_odoo.get_file_mgr();
+    let m1_tf_file_info = file_mgr.borrow().get_file_info(&module1_test_file).unwrap();
+    // Use get_file_info().symbol instead of get_file_symbol
+    let Some(m1_tf_file_symbol) = SyncOdoo::get_symbol_of_opened_file(
+        &mut session,
+        &PathBuf::from(&module1_test_file)
+    ) else {
+        panic!("Failed to get file symbol");
+    };
+
+    let m2_tf_file_info = file_mgr.borrow().get_file_info(&module2_test_file).unwrap();
+    // Use get_file_info().symbol instead of get_file_symbol
+    let Some(m2_tf_file_symbol) = SyncOdoo::get_symbol_of_opened_file(
+        &mut session,
+        &PathBuf::from(&module2_test_file)
+    ) else {
+        panic!("Failed to get file symbol");
+    };
+
+    // Helper to get hover markdown string at a given (line, character)
+    let get_definition_locs = |session: &mut SessionInfo, f_sym: &Rc<RefCell<Symbol>>, f_info: &Rc<RefCell<FileInfo>>, line: u32, character: u32| {
```

Comment:
I think get_definition_locs and get_hover should maybe be public to all tests.
Maybe create a test_utils.rs file?

Reply:
Okay I will make it more generic and not a closure and update it